src/main/webapp/user-location.js

```diff
@@ -0,0 +1,57 @@
+const apiKey = 'AIzaSyB5cba6r-suEYL-0E_nRQfXDtT4XW0WxbQ';
+const fetchJson = {
+  'considerIp': 'true',
+};
+const geoCodingUrl = 'https://maps.googleapis.com/maps/api/geocode/json?latlng=';
+const geoLocationUrl = 'https://www.googleapis.com/geolocation/v1/geolocate?key=';
+
+// Given the user's lat and lng from the geoLocation API,
+// getUserState calls to the geoCoding API to reverse geoCode
+// the coordinates and get the user's current state.
+async function getUserState(lat, lng) {
+  const geoCodingFetch = geoCodingUrl + lat + ',' + lng + '&key=' + apiKey;
+  return fetch(geoCodingFetch).then((response) => response.json())
+    .then(function(jsonResponse) {
+      // Traverse all of the addresses and find one that
+      // has administrative_area_level_1 (state) as one of the address
+      // components.Then return the addressComponent.long_name
+      // which will be the state name.
+      const results = jsonResponse.results;
+      let state;
+      results.forEach((address) => {
+        const addressComponents = address.address_components;
+        addressComponents.forEach((component) => {
+          const types = component.types;
+          types.forEach((type) => {
+            if (type === 'administrative_area_level_1') {
```

Comment:
Can't we return it immediately? Is there an advantage in returning the last address state vs  first address state.

Reply:
Returning it inside a forEach threw an error for me. There is also error checking at the end.